### PR TITLE
Update RGFW

### PR DESCRIPTION
**Pull request by ColleagueRiley**
A couple of updates and bug fixes for RGFW and rcore_desktio_RGFW.c

- Cursor holding is less of a hack. RGFW uses raw mouse movement now. 
- A monitor-related bug is fixed 
- name changes for clarity
- HTML5 support included in RGFW.h

### CODE REVIEW

**Comment by raysan5** [comment]
@ColleagueRiley Thanks for the update!

File: src/platforms/rcore_desktop_rgfw.c [hunk 1]
```diff
@@ -1258,6 +1257,16 @@ int InitPlatform(void)
 
     platform.window = RGFW_createWindow(CORE.Window.title, RGFW_RECT(0, 0, CORE.Window.screen.width, CORE.Window.screen.height), flags);
 
+    RGFW_area screenSize = RGFW_getScreenSize();
+    CORE.Window.display.width = screenSize.w;
+    CORE.Window.display.height = screenSize.h;
+    /* 
+        I think this is needed by Raylib now ?
```
Comment (raysan5) [inline src/platforms/rcore_desktop_rgfw.c:1264]: Why?

Comment (ColleagueRiley) [reply src/platforms/rcore_desktop_rgfw.c:1264]: I saw it in the glfw platform, I'm not sure if it's an init function Raylib requires or not.

File: src/platforms/rcore_desktop_rgfw.c [hunk 2]
```diff
@@ -643,7 +645,7 @@ Vector2 GetWindowScaleDPI(void)
 {
     RGFW_monitor monitor = RGFW_window_getMonitor(platform.window);
 
-    return (Vector2){((u32)monitor.scaleX)*platform.window->r.w, ((u32) monitor.scaleX)*platform.window->r.h};
+    return (Vector2){monitor.scaleX, monitor.scaleX};
```
Comment (raysan5) [inline src/platforms/rcore_desktop_rgfw.c:648]: Not following raylib coding conventions, a space after `{` and before `}` should be used

File: src/platforms/rcore_desktop_rgfw.c [hunk 3]
```diff
@@ -483,14 +483,14 @@ void SetWindowIcons(Image *images, int count)
 // Set title for window
 void SetWindowTitle(const char *title)
 {
-    RGFW_window_setName(platform.window, title);
+    RGFW_window_setName(platform.window, (char*)title);
```
Comment (raysan5) [inline src/platforms/rcore_desktop_rgfw.c:486]: Not following raylib coding conventions, raylib uses `(char *)`

File: src/platforms/rcore_desktop_rgfw.c [hunk 4]
```diff
@@ -689,9 +691,8 @@ void EnableCursor(void)
 void DisableCursor(void)
 {
     RGFW_disableCursor = true;
-    
-    // Set cursor position in the middle
-    SetMousePosition(CORE.Window.screen.width/2, CORE.Window.screen.height/2);
+
+    RGFW_window_mouseHold(platform.window, RGFW_AREA(CORE.Window.screen.width / 2, CORE.Window.screen.height / 2));
```
Comment (raysan5) [inline src/platforms/rcore_desktop_rgfw.c:695]: Not following raylib coding conventions, raylib does not use spaces between `*` and `/` operators

**Comment by raysan5** [comment]
Added some minor formatting comments, no worries, I can review them.

**Pull request merged**